angular/src/app/components/book-card/book-card.component.ts

```diff
@@ -8,4 +9,10 @@ import { IBook } from 'src/app/interfaces/global.interface';
 })
 export class BookCardComponent {
   @Input() book!: IBook;
+
+  constructor(private shoppingService: ShoppingCartService) {}
+
+  addBook() {
```

Comment:
Add data type, in this case void
```suggestion
  addBook(): void {
```